### ISSUE TITLE
Incorrect `date` set for recent commits

### ISSUE THREAD

**Issue by jonhoo** [rust-lang-nursery/rustc-perf#331]
All submitted performance results in [`rustc-timing`](https://github.com/rust-lang-nursery/rustc-timing) since [this commit](https://github.com/rust-lang-nursery/rustc-timing/commit/30873c07b125c2bf217a48c3a7511fb82e41a09e) have their `commit.date` set to `2000-01-01T00:00:00+00:00`. The [previous commit](https://github.com/rust-lang-nursery/rustc-timing/commit/68c9fff8db5c6be18dad175c7f698d7cd50d1c23) from the _same day_ has the correct date set. See for example https://github.com/rust-lang-nursery/rustc-timing/commit/b830de582e9d6cfb89d599777762f1ea33ab1c1f, which tested the commit [`18d646d65ef343566ad6b84732eed1ea001d740a`](https://github.com/rust-lang/rust/commit/18d646d65ef343566ad6b84732eed1ea001d740a) from `2019-01-19T03:21:42Z`. That seems incorrect.

**Comment by Mark-Simulacrum** [comment]
https://github.com/rust-lang-nursery/rustc-timing/commit/b830de582e9d6cfb89d599777762f1ea33ab1c1f - this is in fact a try commit
https://github.com/rust-lang-nursery/rustc-timing/commit/74d3e043e9c8d91245fd68f7ba552c6363d14a2c - again, try commit
https://github.com/rust-lang-nursery/rustc-timing/commit/30873c07b125c2bf217a48c3a7511fb82e41a09e - try commit


So I think this is actually correct just a little unusual. Let me know if you disagree and I'll investigate further!

**Comment by jonhoo** [comment]
It seems odd for the `date` of try commits to not be set to the commit's time? Why should try commits be treated differently? Also, is there a way to detect commits that are _not_ try commits?

**Comment by Mark-Simulacrum** [comment]
Any try commit will have the date set to `2000-01-01T00:00:00+00:00`. This is *primarily* for historical reasons (if we want to change this, I'm happy to take a PR).

With regards to "not try commit" logic, simply `date != January 1st, 2000` would be your best bet I think.

**Comment by jonhoo** [comment]
@Mark-Simulacrum It'd be handy if there was a way to detect try results without having to open+parse the file in question. In [`sunrise`](https://github.com/jonhoo/rust-at-sunrise) for example, I just quickly want to find the latest perf result, but that currently requires [walking commits](https://github.com/jonhoo/rust-at-sunrise/blob/ab031db01ff99629a07f479c3722d1a94815333a/src/main.rs#L308L312) that are for [per results files](https://github.com/jonhoo/rust-at-sunrise/blob/ab031db01ff99629a07f479c3722d1a94815333a/src/main.rs#L317), and then opening all the files to check that they [don't contain](https://github.com/jonhoo/rust-at-sunrise/blob/ab031db01ff99629a07f479c3722d1a94815333a/src/main.rs#L329L333) the string `2000-01-01T00:00:00+00:00`. Which is obviously fallible. I _could_ also parse the JSON, but that's even more extra work.

I think ideally the commit message would mention that it's adding a try commit result, or the result file would contain something like `"try":true` to allow the use of `git log -S`.That'd make the repo more friendly to automated scripts :)

**Comment by Mark-Simulacrum** [comment]
Sure, so there's a couple "immediate" options that I can see. If you're willing to use the GitHub API, then I believe you can query the "10" most recent bors commits on rust-lang/rust fairly easily (with https://developer.github.com/v3/repos/commits/#list-commits-on-a-repository). Then you can try and load each of those from the rustc-timing repository directly, or potentially (if you want to avoid the clone entirely) via the raw content ~API (`https://raw.githubusercontent.com/rust-lang-nursery/rustc-timing/master/times/2018-05-20T00%3A49%3A37%2B00%3A00-21ea121de180d293d0d6d065846750367cb27d95-x86_64-unknown-linux-gnu.json`).

If the latter isn't quite something that works for you I would be happy to add an API that essentially gives the JSON for a particular commit -- e.g. `perf.rust-lang.org/perf/raw_data?commit=sha`.

Basically the reason the files are currently almost identical is because the site actually always loads all the files into memory so I haven't spent any time trying to optimize indexing into them. Exposing some index-like interface -- whether through a file or an API on perf.rust-lang.org -- should be feasible.

**Comment by jonhoo** [comment]
Oh, those are some good pointers, thanks! To take a step back, the thing I want to do is as follows:

When a new nightly is released, I have two refs to two commits in `rust-lang/rust`. One for the old nightly, and one for the new. I then want to compare the perf result for the old nightly with that of the newest nightly. However, `rustc-timing` often doesn't yet have perf results for the commit of the latest yet by the time it's released, so instead I want to find the most recent commit that's `> old` and `<= new`. It's probably fine to assume that there are no newer perf results than that of the nightly since I discover new nightlies pretty quickly, so that really just comes down to finding the latest (non-try) perf result. There's usually a perf result available for the exact commit of the old nightly, so let's ignore that.

With the APIs you gave above, I guess walking back through the history of bors commits on `master` on `rust-lang/rust` and then looking those up in `rustc-timing` is probably the best way to go. I'll give that a shot! Now that files in `times/` are just named `commit-HASH-...`, the `raw.githubusercontent.com` trick should work -- it's only if more info was needed that a `raw_data?commit=sha` endpoint would be necessary. It does mean I'll have to expand the hash, but I could probably do that through the GitHub API. I'll report back how that goes!

Thanks for you help :)

**Comment by jonhoo** [comment]
Seems to work beautifully! https://github.com/jonhoo/rust-at-sunrise/commit/4d249b6c2fde8812d9379939a17b432c313bd3b7
Let's see if it still works when the new nightly releases!
